core/src/HIP/Kokkos_HIP_Instance.cpp

```diff
@@ -374,6 +370,10 @@ void HIPInternal::finalize() {
   this->fence("Kokkos::HIPInternal::finalize: fence on finalization");
   was_finalized = true;
 
+  // Once the underlying stream has been synchronized, check if it's still the one used
+  // for the recorded event.
+  HIPInternal::constantMemReusable[m_hipDev].check_if_involved(m_stream);
```

Comment:
We thought here about the need for a "map from execution space instances to underlying stream". It seems such a map would allow us to avoid doing this call "unnecessarily" (i.e. when another instance with the same underlying stream is still alive). But, it also seems that this call has low overhead and that doing this call even when it's not strictly needed "saves" doing the event synchronize later on. And so we didn't implement a map. @dalg24, do you think this is fine, or how should we think about the need for the map?

Reply:
IMO, even a map of `HIPInternal` instances pointing to the same `hipStream_t` would need to be accessed using atomics, so I don't see why we'd need such a map for now.